game/ffmpeg.cc

```diff
@@ -301,31 +301,31 @@ void FFmpeg::avcodec_free_context(AVCodecContext *avctx) {
 	::avcodec_free_context(&avctx);
 }
 
-struct FFmpeg::Packet: public AVPacket {
-	Packet() {
-		av_init_packet(this);
-		this->data = nullptr;
-		this->size = 0;
-	}
-
-	~Packet() { av_packet_unref(this); }
-};
-
 void FFmpeg::handleOneFrame() {
 	bool read_one = false;
 	do {
-		Packet pkt;
-		auto ret = av_read_frame(m_formatContext.get(), &pkt);
+                std::unique_ptr<AVPacket, std::function<void(AVPacket*)>> pkt(av_packet_alloc(), [] (auto *pkt) { av_packet_free(&pkt); });
+		auto ret = av_read_frame(m_formatContext.get(), pkt.get());
 		if(ret == AVERROR_EOF) {
 			// End of file: no more data to read.
 			throw Eof();
 		} else if(ret < 0) {
 			throw Error(*this, ret);
 		}
 
-		if (pkt.stream_index != m_streamId) continue;
-
-		decodePacket(pkt);
+		if (pkt->stream_index != m_streamId) continue;
+
+                ret = avcodec_send_packet(m_codecContext.get(), pkt.get());
+                if(ret == AVERROR_EOF) {
+                        // End of file: no more data to read.
+                        throw Eof();
+                } else if(ret == AVERROR(EAGAIN)) {
+                        // no room for new data, need to get more frames out of the decoder by
+                        // calling avcodec_receive_frame()
```

Comment:
Let's log something here or remove this else if clause

Reply:
this is done exactly not to log anything as it is "normal" behaviour to have no room left. The EAGAIN needs to be handled separately of errors

Reply:
Alright but since we currently do nothing within this it's just an empty block which can be removed then right? As we're not throwing anything or doing anything. You could move it then to the last else if
`else if (ret < 0 && ret !=  AVERROR(EAGAIN))

Reply:
I did this to separate the flow and add specific comment to explain this behaviour, but if you think I should change it, I will.

Reply:
hmmm i can see the reasoning for this, aight leave it like it is :)